tests/protocol/test_wire.py

```diff
@@ -815,6 +815,41 @@ def http_get_handler(url, *_, **kwargs):
             self.assertTrue(self.is_host_plugin_extension_artifact_request(urls[3]), "The retry request should have been over the host channel")
             self.assertEquals(HostPluginProtocol.is_default_channel(), False, "The default channel should not have changed")
 
+    def test_upload_logs_should_not_refresh_plugin_when_first_attempt_succeeds(self):
+        def http_put_handler(url, *_, **__):
+            if protocol.get_endpoint() in url and url.endswith('/vmAgentLog'):
```

Comment:
let's create a predicate for this condition in HttpRequestPredicates

Reply:
seems like you added it, but did not use it

Reply:
Thanks, updated.